acid/features/zuul_connector/connector.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from contextlib import contextmanager
+import shlex
+import os
+
+import paramiko
+
+from .exceptions import ZuulConnectorFailed
+
+
+class ZuulConnector:
+    def __init__(self, host, username, user_key_file, host_keys_file,
+                 tenant, trigger, project):
+        self.user_key = paramiko.RSAKey.from_private_key_file(user_key_file)
+        self.host = host
+        self.username = username
+        self.host_pub_key = host_keys_file
+
+        self.tenant = tenant
+        self.trigger = trigger
+        self.project = project
+        self._client = self._prepare_client(self.host_pub_key)
+
+    @contextmanager
+    def ssh_connection(self):
+        try:
+            self._client.connect(hostname=self.host, username=self.username,
+                                 pkey=self.user_key)
+            yield self._client
+        except paramiko.ssh_exception.SSHException as exc:
+            raise ZuulConnectorFailed(f"{type(exc).__name__}: {exc}")
```

Comment:
If you have to raise exception based on other exception always raise `from None` or `from exc`
https://www.python.org/dev/peps/pep-0409/
https://stackoverflow.com/a/24752607
```python
try:
    self._client.connect(hostname=self.host, username=self.username, pkey=self.user_key)
    yield self._client
except paramiko.ssh_exception.SSHException as exc:
    raise ZuulConnectorFailed() from exc
finally:
    self._client.close()
```